### PR TITLE
create page

### PR DESCRIPTION
**Pull request by mndys**

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -14,105 +14,151 @@
     <header class="header">
       <h1>riddle me this…</h1>
     </header>
+    <main class="">
```
Comment (badabam) [inline src/index.html]: Please remove that empty class attribute.

File: src/index.html [hunk 2]
```diff
@@ -14,105 +14,151 @@
...
+      <section class="page app-grid">
+        <section class="card" data-js="card">
+          <i class="card__bookmark fa fa-3x fa-bookmark" data-js="bookmark"></i>
+          <h2>
+            How do you make the number 7 even without addition, subtraction,
+            multiplication, or division?
+          </h2>
+          <details>
+            <summary class="card__show-answer">Show Answer</summary>
```
Comment (badabam) [inline src/index.html]: I would prefer to be a button as this is semantically not really a summary of the text below.

File: src/index.html [hunk 3]
```diff
@@ -14,105 +14,151 @@
...
+            <li class="card__tag-list__item">Tag 1</li>
+            <li class="card__tag-list__item">Tag 2</li>
+            <li class="card__tag-list__item">Tag 3</li>
+            <li class="card__tag-list__item">Tag 4</li>
+          </ul>
+        </section>
       </section>
+      <section class="page create-page hidden">
+        <form class="create-page__form">
+          <label class="create-page__form__label"
```
Comment (badabam) [inline src/index.html]: There should not be nested elements in BEM. You are doing "BEEM" here (two Elements). 

Better start a new component "`form`" that has a "`form__label`". No reference for the page type needed.

File: src/index.html [hunk 4]
```diff
@@ -14,105 +14,151 @@
...
+            ><h2 class="create-page__form__heading">Your question:</h2></label
+          >
+          <textarea
+            class="create-page__form__add-question"
+            maxlength="100"
+            placeholder="make it a tough one!"
```
Comment (badabam) [inline src/index.html:146]: Great placeholder Text!

File: src/index.html [hunk 5]
```diff
@@ -14,105 +14,151 @@
...
+            required
+            rows="4"
+            spellcheck="true"
```
Comment (badabam) [inline src/index.html]: Good idea!

File: src/js/index.js [hunk 6]
```diff
@@ -13,6 +13,74 @@ const navBookmark = document.querySelector('[data-js=navBookmark]')
 const navCreate = document.querySelector('[data-js=navCreate]')
 const navSettings = document.querySelector('[data-js=navSettings]')
 
+// Get main sections (^= pages)
```
Comment (badabam) [inline src/js/index.js]: I like how you try to guide us with that comments. We will aim for self explanatory code that does not need as many comments in the future. Stay tuned!

File: src/js/index.js [hunk 7]
```diff
@@ -13,6 +13,74 @@ const navBookmark = document.querySelector('[data-js=navBookmark]')
...
+  '.create-page__form__add-answer__counter'
+)
+const tagCounter = document.querySelector(
+  '.create-page__form__add-tags__counter'
+)
+
+// Get button
+const formButton = document.querySelector('.create-page__form__button')
+
+/* ALPHA TESTING
```
Comment (badabam) [inline src/js/index.js]: Please move experimental code to separate branches. You can merge or delete them later, but they would not spam your production code.

File: src/js/index.js [hunk 8]
```diff
@@ -13,6 +13,74 @@ const navBookmark = document.querySelector('[data-js=navBookmark]')
...
+  return tagCounter.textContent.match(/,/g)
+  ? (tagCounter.textContent = tagCounter.textContent.match(/,/g).length - 5)
+  : (tagCounter.textContent = 5)
+})*/
+
+/* Functions */
+
+// Prevent form from submitting & delete all input
+form.addEventListener('keyup', function (event) {
+  if (event.key === 'Enter') {
```
Comment (badabam) [inline src/js/index.js]: You show quite some knowledge. But this is done by using the Form element and the submit event already. No need for this extra event handling.

File: src/js/index.js [hunk 9]
```diff
@@ -13,6 +13,74 @@ const navBookmark = document.querySelector('[data-js=navBookmark]')
...
+})
+
+// Add counters to textareas
+
+questionArea.addEventListener('input', () => {
+  questionCounter.textContent =
+    questionArea.maxLength - questionArea.value.length
+})
+
+answerArea.addEventListener('input', () => {
```
Comment (badabam) [inline src/js/index.js]: You could use the same logic for counting the question and answer lengths. Just like we did with multiple cards and their answer button logic.

File: src/styles/_base.css [hunk 10]
```diff
@@ -2,6 +2,10 @@
   box-sizing: border-box;
 }
 
+html {
+  height: -webkit-fill-available;
```
Comment (badabam) [inline src/styles/_base.css]: Do you mind explaining that in a comment? I don't know, what it does here.

File: src/styles/_base.css [hunk 11]
```diff
@@ -12,15 +16,50 @@ body {
   margin: 0;
 }
 
+main {
+  display: grid;
+  grid-template-columns: repeat(auto-fit, 375px);
```
Comment (badabam) [inline src/styles/_base.css]: Why `375`?

File: src/styles/_base.css [hunk 12]
```diff
@@ -12,15 +16,50 @@ body {
...
 h2,
 h3 {
   line-height: 1.3;
 }
 
-input,
+h2 {
+  display: block;
+  font-size: 1.5em;
+  margin-block-start: 0.83em;
```
Comment (badabam) [inline src/styles/_base.css]: I love your usage of direction independent css properties.

File: src/styles/_base.css [hunk 13]
```diff
@@ -12,15 +16,50 @@ body {
...
+  padding: 10px 20px;
+  display: block;
+  margin: 28px auto;
+  color: whitesmoke;
+  text-transform: uppercase;
+  text-decoration: none;
+  background: var(--color--imperial-red);
+  border-radius: 5px;
+  border: 2px solid var(--color--amaranth-red);
+  transition: all 0.4s ease 0s;
```
Comment (badabam) [inline src/styles/_base.css]: What is this last `0s` for? Also: `s` might not be needed.

File: src/styles/_base.css [hunk 14]
```diff
@@ -12,15 +16,50 @@ body {
...
+  text-shadow: 1px 1px 3px black;
+  box-shadow: 1px 1px 3px 0 black;
```
Comment (badabam) [inline src/styles/_base.css]: That last `0` could be omitted.

File: src/styles/_base.css [hunk 15]
```diff
@@ -34,3 +73,11 @@ li {
   padding: 2px 7px;
   border-radius: 5px;
 }
+
+textarea,
+input {
+  resize: none;
```
Comment (badabam) [inline src/styles/_base.css:77]: Why not for the textareas?

File: src/styles/_base.css [hunk 16]
```diff
@@ -34,3 +73,11 @@ li {
...
+  width: 100%;
+  font: inherit;
```
Comment (badabam) [inline src/styles/_base.css:79]: Good!

File: src/styles/components/_create-page__form.css [hunk 17]
```diff
@@ -0,0 +1,46 @@
+.create-page__form {
+  max-width: 360px;
+  padding: 18px;
+  margin: 0 auto;
+  display: relative;
+}
+::placeholder {
```
Comment (badabam) [inline src/styles/components/_create-page__form.css:5]: Nice!

File: src/styles/components/_create-page__form.css [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+  text-shadow: 2px 1px 3px black;
+}
+
+.create-page__form__label {
+  display: block;
+}
+
+.create-page__form__add-answer__counter,
+.create-page__form__add-question__counter,
+.create-page__form__add-tags__counter {
```
Comment (badabam) [inline src/styles/components/_create-page__form.css]: Please consider a component `counter` that you use three times. Don't recreate the nesting of your html elements as a class name.

File: src/styles/components/_create-page__form.css [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+  position: relative;
+  font-size: 0.7em;
+  color: var(--color--grey);
+  top: -28px;
+  left: 297px;
```
Comment (badabam) [inline src/styles/components/_create-page__form.css]: Something like `right: 10px` would be preferable. Does not break that easy.

File: src/styles/components/_create-page__form.css [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+  width: 19px;
+  height: 0;
```
Comment (badabam) [inline src/styles/components/_create-page__form.css:30]: Why is that needed?

File: src/styles/components/_create-page__form.css [hunk 21]
```diff
@@ -0,0 +1,46 @@
...
+  text-align: right;
+}
+
+.create-page__form__add-answer__counter {
+  top: -28px;
+  left: 297px;
```
Comment (badabam) [inline src/styles/components/_create-page__form.css]: Isn't all that already in the ruleset above?

**Review by badabam — CHANGES_REQUESTED** [review]
You do a lot of great things here that show your skills. Please have another look on how to write good BEM classes, though. 

I have some questions here and there. I would love to here from you about them. 

👍

File: src/index.html (hunk 3 above)
Comment (mndys) [reply src/index.html]: I got confused because, in my mind, a list item cannot exist without the enclosing list. Therefore I thought I had to nest them which is reflected in the class name. I have since learned that we're using BEM precisely to not have nested elements...😅 I'm not sure I have it right now, but I suppose I just need some practice.

File: src/js/index.js (hunk 8 above)
Comment (mndys) [reply src/js/index.js]: I added this because pressing "Enter" in the textarea-fields gave me a new line rather than submit the form. Have I missed a simpler way here? 🤔

File: src/styles/_base.css (hunk 10 above)
Comment (mndys) [reply src/styles/_base.css]: I read [here](https://css-tricks.com/css-fix-for-100vh-in-mobile-webkit/) that it would fix a display issue on iOS (I can have either the header or the footer fully visible, never both at the same time), but it didn't.

File: src/js/index.js (hunk 9 above)
Comment (mndys) [reply src/js/index.js]: I tried but then I ran into problems with the tag-counter which needs to count comma separated values rather than characters and I gave up 😅

File: src/styles/_base.css (hunk 11 above)
Comment (mndys) [reply src/styles/_base.css]: I tried it out in the browser and thought it was a good size for several columns as well as one on mobile view. Dev Tools says it's the width of the iphone display, so I went with it. With paddings and margins it's probably shorter, isn't it?

File: src/styles/_base.css (hunk 13 above)
Comment (mndys) [reply src/styles/_base.css]: I'm not sure if you can even notice the delay or lack thereof. I played around on a transition-generator website and liked the settings

File: src/styles/_base.css (hunk 15 above)
Comment (mndys) [reply src/styles/_base.css:77]: I thought it might wreck the layout and shouldn't be necessary.

File: src/styles/components/_create-page__form.css (hunk 20 above)
Comment (mndys) [reply src/styles/components/_create-page__form.css:30]: I set the height because the div shouldn't use up space in its place of origin when I moved it into the textarea and the width is set to 19px so I can properly align the counters to be even, whether 2 or 3 digits are on display.

File: src/index.html [hunk 22]
```diff
@@ -14,105 +14,142 @@
...
+          <i class="card__bookmark fa fa-3x fa-bookmark" data-js="bookmark"></i>
+          <h2>
+            How do you make the number 7 even without addition, subtraction,
+            multiplication, or division?
+          </h2>
+          <button class="card__show-answer">Show Answer</button>
+          <p class="card__show-answer__answer text-center hidden">
+            Drop the "s"… 😅
+          </p>
+          <ul class="card__tag-list flex-se">
```
Comment (badabam) [inline src/index.html]: Too much connection between the card component and the tag-list. I'd prefer just having `tag-list` here and not mentioning the `card` again as it is not important for the tag-list to be part of the card.

Below you can use `tag-list__item` then, which is correct BEM.

File: src/index.html [hunk 23]
```diff
@@ -14,105 +14,142 @@
...
-          <li class="card__tag-list__item">Tag 2</li>
-          <li class="card__tag-list__item">Tag 3</li>
-          <li class="card__tag-list__item">Tag 4</li>
-        </ul>
-      </section>
+        <section class="card" data-js="card">
+          <i class="card__bookmark fa fa-3x fa-bookmark" data-js="bookmark"></i>
+          <h2>What can you catch but never throw?</h2>
+          <button class="card__show-answer">Show Answer</button>
+          <p class="card__show-answer__answer text-center hidden">A cold 🤧</p>
```
Comment (badabam) [inline src/index.html]: Also here: too much nesting and connecting between the elements. Instead of `card__show-answer__answer`, you should only write `card__answer`. There is never a `onething__secondthing__thirdthing`, only `component__element`. For the styling of the "answer" it is irrelevant that there is a button to show it. But it's relevant that the answer is part of the card: if you delete the button, the answer does not care. If you delete the card, there would be no answer anymore as it can't (or is unlikely to) exist without the card.

File: src/js/index.js (hunk 8 above)
Comment (badabam) [reply src/js/index.js]: Ah, I did not think of it being `textarea`s. You are right: without your code here, it would not submit the form on Enter. 

But if we use textareas, we want them to have multiple lines and not submit on Enter – so better get rid of this.

File: src/js/index.js (hunk 9 above)
Comment (badabam) [reply src/js/index.js]: Wow, that's an ambitious thing to try (at least with what we learned so far). I like that you tried it. Let's have a talk about that tomorrow with the other students. Please remind me!

File: src/styles/_base.css (hunk 10 above)
Comment (badabam) [reply src/styles/_base.css]: Please try it the way I shared with you in [this codepen](https://codepen.io/neuefische/pen/PoZeqzQ?editors=0100).

File: src/styles/_base.css (hunk 11 above)
Comment (badabam) [reply src/styles/_base.css]: Please add a comment to your code that explains the number (if you are still sure you want to use it).

File: src/styles/_base.css (hunk 13 above)
Comment (badabam) [reply src/styles/_base.css]: The last `0s` does nothing. Please remove it.

File: src/styles/_base.css (hunk 15 above)
Comment (badabam) [reply src/styles/_base.css:77]: Ok.

File: src/styles/components/_create-page__form.css (hunk 20 above)
Comment (badabam) [reply src/styles/components/_create-page__form.css:30]: You should use `position: absolute` for the counters so they don't use up space. The width of `19px` also sounds a bit _hacky_. Isn't aligning them (maybe to the right) enough?

**Review by badabam — COMMENTED** [review]
I answered a bit in our conversation about your code. Maybe we should talk about that on zoom to speed things up. But first let's see how things fit after considering my remarks.
